lib/screen/playlists/list_playlists/list_playlists.dart

```diff
@@ -77,68 +78,116 @@ class _ListPlaylistsScreenState extends State<ListPlaylistsScreen>
-                              context,
-                              AppRouter.viewPlayListPage,
-                              arguments: ViewPlaylistScreenPayload(
-                                  playListModel: item),
-                            ));
-                            final tokenIds = item.tokenIDs;
-                            if (tokenIds != null && tokenIds.isNotEmpty) {
-                              final bloc = injector.get<CanvasDeviceBloc>();
-                              final controllingDevice =
-                                  bloc.state.controllingDevice;
-                              if (controllingDevice != null) {
-                                final duration = speedValues.values.first;
-                                final List<PlayArtworkV2> castArtworks =
-                                    tokenIds
-                                        .map((e) => PlayArtworkV2(
-                                              token: CastAssetToken(id: e),
-                                              duration: duration.inMilliseconds,
-                                            ))
-                                        .toList();
-                                bloc.add(CanvasDeviceChangeControlDeviceEvent(
-                                    controllingDevice, castArtworks));
-                              }
-                            }
-                          });
-                    },
-                    separatorBuilder: (context, index) =>
-                        const SizedBox(width: 10),
-                  ),
-                ),
+                if ((widget.playlists.value?.length ?? 0) >=
+                    playlistNumberBreakpoint)
+                  playlistHorizontalGridView(context, playlists)
+                else
+                  playlistListView(context, playlists)
               ],
             ),
           );
         },
       );
+
+  Widget playlistListView(BuildContext context, List<PlayListModel> playlists) {
+    const height = PlaylistItem.height;
+    return SizedBox(
+      height: height,
+      child: ListView.separated(
+        scrollDirection: Axis.horizontal,
+        itemCount: playlists.length + 1,
+        itemBuilder: (context, index) {
+          if (index == 0) {
+            if (widget.filter.isNotEmpty) {
+              return const SizedBox();
+            }
+            return AddPlayListItem(
+              onTap: () {
+                widget.onAdd();
+              },
+            );
+          }
+          final item = playlists[index - 1];
+          return SizedBox(
+            height: PlaylistItem.height,
+            child: PlaylistItem(
+                playlist: item,
+                onSelected: () {
+                  onPlaylistTap(item);
+                }),
+          );
+        },
+        separatorBuilder: (context, index) => const SizedBox(width: 15),
+      ),
+    );
+  }
+
+  Widget playlistHorizontalGridView(
```

Comment:
why not always use gridView, just change the crossAxisCount
Also, mark it as private as possible